json/src/fs2/data/json/internal/Renderer.scala

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2020 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package fs2
+package data
+package json
+package internals
+
+import scala.annotation.{tailrec, switch}
+
+private[json] class Renderer(pretty: Boolean, resetOnChunk: Boolean, indent: String)
+    extends Collector.Builder[Token, String] {
+
+  import Renderer._
+
+  private val builder = new StringBuilder
+
+  private var level = 0
+
+  private var newline = false
+
+  private var comma = false
+
+  private def separator(comma: Boolean): Unit = {
+    if (comma)
+      builder.append(',')
+  }
+
+  private def indentation(newline: Boolean, level: Int): Unit =
+    if (newline) {
+      builder.append('\n')
+      builder.append(indent * level)
+    }
+
+  private def jsonString(s: String, key: Boolean): Unit = {
+    @tailrec
+    def loop(idx: Int): Unit =
```

Comment:
@ybasket I also fixed escaping, but for the sake of "you never know what you do on a Saturday morning, so better safe than sorry", can you quickly check this new version?

Reply:
It's also Saturday morning for me ;)

But looks all reasonable!